opentracing-util/src/main/java/io/opentracing/util/AutoFinisher.java

```diff
@@ -0,0 +1,53 @@
+package io.opentracing.util;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * XXX: fix comment
+ * {@link AutoFinisher} is an {@link ActiveSpan} wrapper that automatically {@link Span#finish()}es the underlying
```

Comment:
nit: could avoid conjugating method names, e.g. "wrapper that automatically calls `finish()` on the underlying...`